src/app/components/alert/alert.component.ts

```diff
@@ -9,45 +10,49 @@ import { v4 as uuid } from "uuid";
   standalone: true,
   templateUrl: "./alert.component.html",
   styleUrl: "./alert.component.scss",
-  encapsulation: ViewEncapsulation.None,
-  imports: [CommonModule]
+  imports: [CommonModule],
+  hostDirectives: [
+    {
+      directive: TitleDirective,
+      inputs: ["title"],
+    },
+  ]
 })
 export class AlertComponent implements OnInit {
+  // dependency injection
+  private titleDirective = inject(TitleDirective);
```

Comment:
Moving forward, we will use this style of dependency injection everywhere, which renders the use of empty constructors as useless.